### PR TITLE
[cli] Include config from ESC in `pulumi config`

### PR DESCRIPTION
**Pull request by pgavlin**
These changes include any configuration values sourced from a stack's ESC environment to the output of `pulumi config`.

These changes also add an `ENVIRONMENT` block to the output of `pulumi config` for stacks that reference environments. This block shows the definition of the stack's environment.

Finally, these changes add a warning to `pulumi config` if the stack's ESC environment does not define any of the `environmentVariables`, `files`, or `pulumiConfig` properties.

### CODE REVIEW

**Comment by pgavlin** [comment]
Before: https://asciinema.org/a/VvTbjTeVbtwUmz18vITJzg4Q5
After: https://asciinema.org/a/I9hOAT17I95K1g3D037TzUew0

**Comment by pgavlin** [comment]
Ping.

**Comment by pgavlin** [comment]
(I have several other changes that are blocked on this functionality)

File: pkg/cmd/pulumi/config.go [hunk 1]
```diff
@@ -917,6 +947,38 @@ func listConfig(ctx context.Context,
...
+			if len(environ) != 0 {
+				environRows := make([]cmdutil.TableRow, len(environ))
+				for i, kvp := range environ {
+					key, value, _ := strings.Cut(kvp, "=")
+					environRows[i] = cmdutil.TableRow{Columns: []string{key, value}}
+				}
+
+				fmt.Println()
+				printTable(cmdutil.Table{
+					Headers: []string{"ENVIRONMENT VARIABLE", "VALUE"},
```
Comment (justinvp) [inline pkg/cmd/pulumi/config.go:969]: Seems like there could be some confusion over system envvars, but I don't have a great alternative suggestion if this is indeed the terminology.

**Review by justinvp — APPROVED** [review]
Looks reasonable to me. Apologies for the delay!

File: pkg/cmd/pulumi/config.go (hunk 1 above)
Comment (Frassle) [reply pkg/cmd/pulumi/config.go:969]: Does ESC actually call these variables or keys/paths?

Comment (pgavlin) [reply pkg/cmd/pulumi/config.go:969]: These are actual system environment variables that will be set during `up`/`preview`/etc. using values plucked from the `environmentVariables` key in the stack's environment.

For example, if my stack references an environment that looks like this:

```yaml
# Here's the environment config
values:
    aws:
        creds:
            fn::open::aws-login:
                oidc:
                    duration: 1h
                    roleArn: <role-arn>
                    sessionName: pulumi-session
    environmentVariables:
        AWS_ACCESS_KEY_ID: ${aws.creds.accessKeyId}
        AWS_SECRET_ACCESS_KEY: ${aws.creds.secretAccessKey}
        AWS_SESSION_TOKEN: ${aws.creds.sessionToken}
```

Then we will set the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` environment variables during Pulumi operations. With these changes we would display those environment variables in the output of `pulumi config` as well.

Comment (justinvp) [reply pkg/cmd/pulumi/config.go:969]: Ah, ok.

**Pull request merged**